ui/src/app/views/workspace-edit/component.ts

```diff
@@ -122,31 +125,71 @@ export class WorkspaceEditComponent implements OnInit {
     this.savingWorkspace = false;
   }
 
+  validateForm(): boolean {
+    if (this.savingWorkspace) {
+      return false;
+    }
+    this.nameNotEntered = isBlank(this.workspace.name);
+    if (this.nameNotEntered) {
+      return false;
+    }
+    return true;
+  }
+
+  addWorkspace(): void {
+    if (!this.validateForm()) {
+      return;
+    }
+    this.savingWorkspace = true;
+    this.errorHandlingService.retryApi(
+      this.workspacesService.createWorkspace(this.workspace))
+      .subscribe(
+        () => {
+          this.navigateBack();
+        },
+        (error) => {
+          this.workspaceCreationError = true;
+        });
+  }
+
   updateWorkspace(): void {
-    if (!this.savingWorkspace) {
-      if (isBlank(this.workspace.name)) {
-        this.nameNotEntered = true;
-      } else {
-        this.savingWorkspace = true;
-        this.nameNotEntered = false;
-        this.errorHandlingService.retryApi(this.workspacesService.updateWorkspace(
-            this.oldWorkspaceNamespace,
-            this.oldWorkspaceName,
-            this.workspace))
-          .subscribe(
-            () => {
-              this.navigateBack();
-            },
-            (error) => {
-              if (error.status === 409) {
-                this.workspaceUpdateConflictError = true;
-              } else {
-                this.workspaceUpdateError = true;
-              }
-            });
-      }
+    if (!this.validateForm()) {
+      return;
     }
+    this.savingWorkspace = true;
+    this.errorHandlingService.retryApi(this.workspacesService.updateWorkspace(
+      this.oldWorkspaceNamespace,
+      this.oldWorkspaceName,
+      this.workspace))
+      .subscribe(
+        () => {
+          this.navigateBack();
+        },
+        (error) => {
+          if (error.status === 409) {
+            this.workspaceUpdateConflictError = true;
+          } else {
+            this.workspaceUpdateError = true;
+          }
+        });
   }
+
+  cloneWorkspace(): void {
+    if (!this.validateForm()) {
+      return;
+    }
+    this.savingWorkspace = true;
+    this.errorHandlingService.retryApi(this.workspacesService.cloneWorkspace(
+      this.oldWorkspaceNamespace,
+      this.oldWorkspaceName, {
+        workspace: this.workspace,
+      }))
+      .subscribe(
+        (resp: CloneWorkspaceResponse) => {
```

Comment:
Not a suggestion, but the scope of usage of this variable is narrow-enough that even `r` would be fine with me.

Reply:
Indeed, done. You might be a fan of the Go style guide